components/support/sql/src/open_database.rs

```diff
@@ -118,6 +120,38 @@ pub fn open_database_with_flags<CI: ConnectionInitializer, P: AsRef<Path>>(
     })
 }
 
+/// Open a read-write database
+pub fn open_read_write_database<CI: ConnectionInitializer, P: AsRef<Path>>(
+    path: P,
+    connection_initializer: &CI,
+) -> Result<Connection> {
+    open_database_with_flags(
+        path,
+        OpenFlags::SQLITE_OPEN_URI
+            | OpenFlags::SQLITE_OPEN_NO_MUTEX
+            | OpenFlags::SQLITE_OPEN_CREATE
+            | OpenFlags::SQLITE_OPEN_READ_WRITE,
+        connection_initializer,
+    )
+}
```

Comment:
I also think this function and the next are nice to add.  I'd rather handle the SQLite flags in this crate than in a component.

Reply:
I wonder if later we should just return the flags, so consumers with a slightly different requirement can still levergage this?

Reply:
I like that idea, just updated the code to match.